Code/Framework/AzCore/AzCore/DOM/DomPath.cpp

```diff
@@ -53,17 +53,20 @@ namespace AZ::Dom
 
     bool PathEntry::operator==(size_t value) const
     {
-        return IsIndex() && GetIndex() == value;
+        const size_t* internalValue = AZStd::get_if<size_t>(&m_value);
+        return internalValue == nullptr ? false : (*internalValue) == value;
```

Comment:
Uh there is no need of a ternary operator in a function that returns a bool.

```suggestion
        return internalValue != nullptr  && *internalValue == value;
```

This is the @lsemp3d pet-peeve

Reply:
lol I knew I'd accidentally check one of these in one day, will fix